internal/clients/gateway_test.go

```diff
@@ -0,0 +1,267 @@
+	assert.Equal(t, "2", deployTx.ConstructorCalldata[2].Text(16))
+	assert.Equal(t, "25b9dbdab19b190a556aa42cdfbc07ad6ffe415031e42a8caffd4a2438d5cc3", deployTx.ConstructorCalldata[3].Text(16))
+	assert.Equal(t, "0", deployTx.ConstructorCalldata[4].Text(16))
+	assert.Equal(t, "DEPLOY_ACCOUNT", deployTx.Type)
+}
+
+func TestL1HandlerTransactionUnmarshal(t *testing.T) {
+	handlerJson := []byte(`{
+      "transaction_hash":"0x218adbb5aea7985d67fe49b45d44a991380b63db41622f9f4adc36274d02190",
+      "version":"0x0",
+      "contract_address":"0x73314940630fd6dcda0d772d4c972c4e0a9946bef9dabf4ef84eda8ef542b82",
+      "entry_point_selector":"0x2d757788a8d8d6f21d1cd40bce38a8222d70654214e96ff95d8086e684fbee5",
+      "nonce":"0x1654d",
+      "calldata":[
+         "0xae0ee0a63a2ce6baeeffe56e7714fb4efe48d419",
+         "0x218559e75713ca564d6eaf043b73388e9ac7c2f459ef8905988052051d3ef5e",
+         "0x2386f26fc10000",
+         "0x0"
+      ],
+      "type":"L1_HANDLER"
+   }`)
+	var handlerTx Transaction
+	err := json.Unmarshal(handlerJson, &handlerTx)
+	if err != nil {
+		t.Error(err)
+	}
+
+	assert.Equal(t, "218adbb5aea7985d67fe49b45d44a991380b63db41622f9f4adc36274d02190", handlerTx.Hash.Text(16))
+	assert.Equal(t, "0", handlerTx.Version.Text(16))
+	assert.Equal(t, "73314940630fd6dcda0d772d4c972c4e0a9946bef9dabf4ef84eda8ef542b82", handlerTx.ContractAddress.Text(16))
+	assert.Equal(t, "2d757788a8d8d6f21d1cd40bce38a8222d70654214e96ff95d8086e684fbee5", handlerTx.EntryPointSelector.Text(16))
+	assert.Equal(t, "1654d", handlerTx.Nonce.Text(16))
+	assert.Equal(t, 4, len(handlerTx.Calldata))
+	assert.Equal(t, "ae0ee0a63a2ce6baeeffe56e7714fb4efe48d419", handlerTx.Calldata[0].Text(16))
+	assert.Equal(t, "218559e75713ca564d6eaf043b73388e9ac7c2f459ef8905988052051d3ef5e", handlerTx.Calldata[1].Text(16))
+	assert.Equal(t, "2386f26fc10000", handlerTx.Calldata[2].Text(16))
+	assert.Equal(t, "0", handlerTx.Calldata[3].Text(16))
+	assert.Equal(t, "L1_HANDLER", handlerTx.Type)
+}
+
+func TestBlockUnmarshal(t *testing.T) {
+	blockJson, err := os.ReadFile("block_11817.json")
+	if err != nil {
+		t.Error(err)
+	}
+
+	var block Block
+	err = json.Unmarshal(blockJson, &block)
+	if err != nil {
+		t.Error(err)
+	}
+
+	assert.Equal(t, "24c692acaed3b486990bd9d2b2fbbee802b37b3bd79c59f295bad3277200a83", block.Hash.Text(16))
+	assert.Equal(t, "3873ccb937f14429b169c654dda28886d2cc2d6ea17b3cff9748fe5cfdb67e0", block.ParentHash.Text(16))
+	assert.Equal(t, uint64(11817), block.Number)
+	assert.Equal(t, "3df24be7b5fed6b41de08d38686b6142944119ca2a345c38793590d6804bba4", block.StateRoot.Text(16))
+	assert.Equal(t, "ACCEPTED_ON_L2", block.Status)
+	assert.Equal(t, "27ad16775", block.GasPrice.Text(16))
+	assert.Equal(t, 52, len(block.Transactions))
+	assert.Equal(t, 52, len(block.Receipts))
+	assert.Equal(t, uint64(1669465009), block.Timestamp)
+	assert.Equal(t, "0.10.1", block.Version)
+}
```

Comment:
I think we need tests for `GetClassDefinition` as well

Reply:
I must have forgotten to push the latest changes I amended, fixed